pocs/dome/abstract_serial_dome.py

```diff
@@ -0,0 +1,81 @@
+from pocs import dome
+from pocs.utils import error
+from pocs.utils import rs232
+
+
+class AbstractSerialDome(dome.AbstractDome):
+    """Abstract base class for controlling a dome via a serial connection.
+
+    Takes care of a single thing: configuring the connection to the device.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize an AbstractSerialDome.
+
+        Creates a serial connection to the port indicated in the config.
+        """
+        super().__init__(*args, **kwargs)
+
+        # Get config info, e.g. which port (e.g. /dev/ttyUSB123) should we use?
+        cfg = self._dome_config
+        self._port = cfg.get('port')
+        if not self._port:
+            msg = 'No port specified in the config for dome: {}'.format(cfg)
+            self.logger.error(msg)
+            raise error.DomeNotFound(msg=msg)
+
+        baudrate = int(cfg.get('baudrate', 9600))
+
+        # Setup our serial connection to the given port.
+        self.ser = None
+        try:
+            self.ser = rs232.SerialData(port=self._port, baudrate=baudrate)
+        except Exception as err:
+            raise error.DomeNotFound(err)
+
+    def __del__(self):
+        try:
+            if self.ser:
+                self.ser.disconnect()
+        except NameError:
+            pass
+
+    @property
+    def is_connected(self):
+        """True if connected to the hardware or driver."""
+        if self.ser:
+            return self.ser.is_connected
```

Comment:
Seems like we should be setting the `self._is_connected` property and then returning that. Not that someone should be messing with the private variable, but this would get it out of sync.

Reply:
I've decided that it doesn't make great sense to put the _is_connected in the base class, AbstractDome, in part because of issues exactly like this. So, I'm inclined to respond to this feedback by stripping it out from AbstractDome, and implementing it in the sub-classes as appropriate to the means to establishing that connection.